src/main/java/core/interfaces/IVectorisable.java

```diff
@@ -0,0 +1,20 @@
+package core.interfaces;
+
+import core.actions.AbstractAction;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+
+import java.util.List;
+
+public interface IVectorisable {
+
```

Comment:
I think there are two separate functions here, that should be two separate interfaces:
1) getJSON() - which is an interface implemented by a GameState if it provides a nice JSON representation.
2) the double[] arrays for use in RL. 

For 2) I suggest we use the existing IStateFeatureVector, and add the missing two methods (the normalised version can be a default?). This means that PyTag takes in an IStateFeatureVector as an additional argument with GameState, and uses this to generate the double[] feature representation. We can then plug-and-play and compare different state feature representations for the same game easily.

Reply:
Separated the feature extractor interfaces as suggested. Moved all the functions out of the GameStates and implemented them as separate classes for each game. PyTAG now has a "registry" that is used to check which games implement which interfaces and uses them to convert the game states with the right feature extractors.